### PR TITLE
[WIP] update docker, containerd, advisor

### PR DESCRIPTION
**Pull request by khushbuparakh**
**What type of PR is this?**
fresh versions of docker, containerd and cadvisor

**What this PR does / why we need it**:
Update containerd v1.4.1
Updating docker v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b
Updating Advisor
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by khushbuparakh** [comment]
Closing this PR

**Comment by khushbuparakh** [comment]
https://github.com/kubernetes/kubernetes/pull/96163

File: vendor/modules.txt [hunk 1]
```diff
@@ -2461,10 +2467,10 @@ k8s.io/metrics/pkg/client/custom_metrics/fake
 k8s.io/metrics/pkg/client/custom_metrics/scheme
 k8s.io/metrics/pkg/client/external_metrics
 k8s.io/metrics/pkg/client/external_metrics/fake
-# k8s.io/mount-utils v0.0.0 => ./staging/src/k8s.io/mount-utils
+# k8s.io/mount-utils v0.20.0-alpha.3 => k8s.io/mount-utils v0.20.0-alpha.3
 ## explicit
 k8s.io/mount-utils
-# k8s.io/mount-utils => ./staging/src/k8s.io/mount-utils
+# k8s.io/mount-utils => k8s.io/mount-utils v0.20.0-alpha.3
```
Comment (aojea) [inline vendor/modules.txt:2473]: hmm are you running ?

```
hack/pin-dependency.sh github.com/docker/docker bd33bbf0497b2327516dc799a5e541b720822a4c
hack/update-vendor.sh
```
to do the updates?

Comment (khushbuparakh) [reply vendor/modules.txt:2473]: yeah!

Comment (khushbuparakh) [reply vendor/modules.txt:2473]: && 
hack/pin-dependency.sh github.com/containerd/containerd v1.4.1
hack/pin-dependency.sh github.com/google/cadvisor b31c0ce7a97f1099fd2c94d6bd4d01e3795ceaa5
hack/update-vendor.sh

**Comment by fedebongio** [comment]
/assign @caesarxuchao 
/triage accepted
(thank you Chao)

**Comment by caesarxuchao** [comment]
I was assigned because this is blocking https://github.com/kubernetes/kubernetes/pull/95981, but really the dependencies in question belongs to sig-node.

/remove-sig api-machinery
/sig node
/unassign

Let's try if @cheftako @yujuhong @kubernetes/sig-node-pr-reviews can review or triage.

**Comment by dims** [comment]
@khushbuparakh please see the log in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/96163/pull-kubernetes-dependencies/1323676509969321984

```
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

github.com/Microsoft/go-winio
    pinned:    v0.4.15-0.20190919025122-fc70bd9a86b5
    preferred: v0.4.15-0.20200908182639-5b44b70ab3ab
    hack/pin-dependency.sh github.com/Microsoft/go-winio v0.4.15-0.20200908182639-5b44b70ab3ab
github.com/containerd/ttrpc
    pinned:    v1.0.0
    preferred: v1.0.2
    hack/pin-dependency.sh github.com/containerd/ttrpc v1.0.2
github.com/containerd/typeurl
    pinned:    v1.0.0
    preferred: v1.0.1
    hack/pin-dependency.sh github.com/containerd/typeurl v1.0.1
github.com/opencontainers/runc
    pinned:    v1.0.0-rc91.0.20200707015106-819fcc687efb
    preferred: v1.0.0-rc92
    hack/pin-dependency.sh github.com/opencontainers/runc v1.0.0-rc92
github.com/opencontainers/runtime-spec
    pinned:    v1.0.3-0.20200520003142-237cc4f519e2
    preferred: v1.0.3-0.20200728170252-4d89ac9fbff6
    hack/pin-dependency.sh github.com/opencontainers/runtime-spec v1.0.3-0.20200728170252-4d89ac9fbff6
golang.org/x/sys
    pinned:    v0.0.0-20200622214017-ed371f2e16b4
    preferred: v0.0.0-20201013081832-0aaa2718063a
    hack/pin-dependency.sh golang.org/x/sys v0.0.0-20201013081832-0aaa2718063a
google.golang.org/grpc
    pinned:    v1.27.0
    preferred: v1.27.1
    hack/pin-dependency.sh google.golang.org/grpc v1.27.1

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/docker/go-events
GO111MODULE=on go mod edit -dropreplace github.com/go-ini/ini
GO111MODULE=on go mod edit -dropreplace github.com/gogo/googleapis
All pinned versions of checked dependencies match their preferred version.
```

**Comment by khushbuparakh** [comment]
All pinned versions of checked dependencies match their preferred version.

**Comment by dims** [comment]
@khushbuparakh looks like @aojea needed more updates :)

https://github.com/kubernetes/kubernetes/pull/96100#issuecomment-720501890
https://github.com/kubernetes/kubernetes/pull/96088#issue-513705608

```
containerd/console f1b333f
moby/term 3e73b07
moby/moby 41d4112
```

**Comment by khushbuparakh** [comment]
> @khushbuparakh looks like @aojea needed more updates :)
> 
> [#96100 (comment)](https://github.com/kubernetes/kubernetes/pull/96100#issuecomment-720501890)
> [#96088 (comment)](https://github.com/kubernetes/kubernetes/pull/96088#issue-513705608)
> 
> ```
> containerd/console f1b333f
> docker/term 3e73b07
> docker/docker 41d4112
> ```

```
some issue here with docker/term
Running: go get github.com/docker/term@3e73b07ecbf5dc7b59fcecc783c4988c6b5aa767
github.com/docker/term@3e73b07ecbf5dc7b59fcecc783c4988c6b5aa767: invalid version: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /home/aviuser/go/pkg/mod/cache/vcs/3b7e41d11eb38d69f140d6eb369a7e7958a06190de3eb912d621c8532e214460: exit status 128:
	ERROR: Repository not found.
	fatal: Could not read from remote repository.
```

**Comment by aojea** [comment]
Try using moby/term, and that's the commit with the fix but iirc there are several commit, what I did is to use the latest tag from docker
9c15e82f19b0ad3c5fe8617a8ec2dddc6639f40a

**Comment by khushbuparakh** [comment]
> Try using moby/term, and that's the commit with the fix but iirc there can be multiple commit, so what I did is to use the latest tag from docker
> 9c15e82f19b0ad3c5fe8617a8ec2dddc6639f40a that contains the fixes, and from there you can get the commits to the containers/console and moby/term projects https://github.com/moby/moby/blob/master/vendor.conf

should we test it now?

**Comment by aojea** [comment]
you can test it locally, I did it with current status and still failing
```
$ KUBE_BUILD_PLATFORMS=darwin/amd64 make quick-release
...
stat /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/bazel-bin: no such file or directory
+++ [1105 09:45:14] Building go targets for darwin/amd64:
    cmd/gendocs
    cmd/genkubedocs
    cmd/genman
    cmd/genyaml
    cmd/genswaggertypedocs
    cmd/linkcheck
    vendor/github.com/onsi/ginkgo/ginkgo
    test/e2e/e2e.test
    cluster/images/conformance/go-runner
# k8s.io/kubernetes/vendor/github.com/docker/docker/pkg/term
vendor/github.com/docker/docker/pkg/term/tc.go:13:28: undefined: unix.SYS_IOCTL
vendor/github.com/docker/docker/pkg/term/tc.go:18:28: undefined: unix.SYS_IOCTL
vendor/github.com/docker/docker/pkg/term/termios_bsd.go:24:31: undefined: unix.SYS_IOCTL
vendor/github.com/docker/docker/pkg/term/termios_bsd.go:37:31: undefined: unix.SYS_IOCTL
```

**Comment by dims** [comment]
@aojea @khushbuparakh use the SHA for x/sys from the go.mod in docker/term

**Comment by khushbuparakh** [comment]
> @aojea @khushbuparakh use the SHA for x/sys from the go.mod in docker/term

okay!

**Comment by dims** [comment]
thanks for the PR @khushbuparakh bits and pieces of this landed as part of other PRs.

/close

**Pull request closed without merging**